lib/grn_hash.h

```diff
@@ -246,7 +246,7 @@ struct _grn_hash {
   // unsigned int subrec_size;
   // grn_rec_unit record_unit;
   // grn_rec_unit subrec_unit;
-  // uint8_t arrayp;
+  // uint8_t array;
```

Comment:
I think that this is not a typo. I think that this means "array predicate" (is array or not).

Anyway, we can remove these comment outed codes. We will never use them. We can remove them before or after this PR.